src/app/public/modules/phone-field/phone-field.component.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  OnDestroy
+} from '@angular/core';
+
+require('intl-tel-input/build/js/utils');
+require('intl-tel-input/build/js/intlTelInput');
+
+/**
+ * NOTE: We can not type these due the the current @types/intl-tel-input version having an
+ * undeclared type which causes linting errors.
+ */
+declare var intlTelInputUtils: any;
+declare var intlTelInputGlobals: any;
+
+import {
+  SkyCountryData
+} from './types';
+
+@Component({
+  selector: 'sky-phone-field',
+  templateUrl: './phone-field.component.html',
+  styleUrls: ['./phone-field.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyPhoneFieldComponent implements OnDestroy {
+
+  public countryData: SkyCountryData[];
+
+  public defaultCountryData: SkyCountryData;
+
+  public disabled: boolean;
+
+  public set selectedCountry(newCountry: SkyCountryData) {
+    this._selectedCountry = newCountry;
+
+    if (!this._selectedCountry.placeholder) {
+      this._selectedCountry.placeholder = intlTelInputUtils.getExampleNumber(
+        newCountry.iso2,
+        true,
+        intlTelInputUtils.numberType.FIXED_LINE
+      );
+    }
+
+    this.countryData.splice(this.countryData.indexOf(newCountry), 1);
+
+    let sortedNewCountries = this.countryData
```

Comment:
Would you consider this sort for a good encapsulated method? It's hard to tell, at a glance, what its doing. Something like `alphabeticallySortAndShowDefault()`?